activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb

```diff
@@ -418,9 +418,11 @@ def release_advisory_lock(lock_id) # :nodoc:
       end
 
       def enable_extension(name, **)
-        exec_query("CREATE EXTENSION IF NOT EXISTS \"#{name}\"").tap {
-          reload_type_map
-        }
+        schema, name = name.to_s.split(".").values_at(-2, -1)
+        sql = "CREATE EXTENSION IF NOT EXISTS \"#{name}\""
+        sql << " SCHEMA #{schema}" if schema
```

Comment:
It raises `FrozenError` with Ruby 2.7.

https://buildkite.com/rails/rails/builds/88742#018284e3-fd0c-41cf-83a4-394a837de8bd/1077-1081

```ruby
Error:
ActiveRecord::ConnectionAdapters::PostgreSQLAdapterTest#test_enable_extension:
FrozenError: can't modify frozen String: "CREATE EXTENSION IF NOT EXISTS \"hstore\""
    /rails/activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb:423:in `enable_extension'
    /rails/activerecord/test/cases/adapters/postgresql/postgresql_adapter_test.rb:156:in `test_enable_extension'
```

While the magic comment `frozen_string_literal: true` is enabled, Ruby 2.7 freezes interpolated strings. On the other hand Ruby 3.0 or higher returns self because `sql` is interpolated one, which is not frozen anymore.  Refer to https://bugs.ruby-lang.org/issues/17104 for the details.